src/language-markdown/printer-markdown.js

```diff
@@ -557,6 +557,11 @@ function printTable(path, options, print) {
     contents[0].map(() => 3) // minimum width = 3 (---, :--, :-:, --:)
   );
 
+  const maxWidth = columnMaxWidths.reduce(
+    (preWidth, width) => preWidth + width,
+    0
+  );
+  const greaterThanPrintWidth = maxWidth > options.printWidth;
```

Comment:
You cannot directly use `maxWidth > options.printWidth` here since tables are allowed to be used in nested lists, where their indentations can be varying.

You need to use some [commands](https://github.com/prettier/prettier/blob/master/commands.md) to describe it, for example:

```js
group(ifBreak(compactTable, alignedTable))
```

Reply:
I will add a test for nested tables.

Reply:
sorry，I have encountered some difficulties. Where is the data structure of doc?

Reply:
I'm not sure what did you mean but those commands are basically used to describe what should we print if it fits the print width (flat) or exceeds the print width (break), then we use the algorithm from "A prettier printer" to output the result. (See https://prettier.io/docs/en/technical-details.html, implementations are placed in `src/doc/`.)

Since the feature you'd like to implement requires some implementation details so I wrote a simple example to demonstrate what should it look like:

```js
// test.js

const {
  group,
  ifBreak,
  concat,
  hardline,
  join,
  breakParent
} = require("./src/doc").builders;

const hardlineWithoutBreakParent = hardline.parts[0];

const content = concat([
  breakParent,
  group(
    ifBreak(
      join(
        hardlineWithoutBreakParent,
        Array.from(new Array(5), () => "compact" + "-".repeat(100))
      ),
      join(
        hardlineWithoutBreakParent,
        Array.from(new Array(5), () => "aligned" + "-".repeat(100))
      )
    )
  )
]);

module.exports = {
  languages: [{ name: "test", parsers: ["test"] }],
  parsers: { test: { parse: () => ({}), astFormat: "test" } },
  printers: { test: { print: () => content } }
};
```

```
$ bin/prettier.js --plugin test.js --parser test test.js --print-width 80
compact----------------------------------------------------------------------------------------------------
compact----------------------------------------------------------------------------------------------------
compact----------------------------------------------------------------------------------------------------
compact----------------------------------------------------------------------------------------------------
compact----------------------------------------------------------------------------------------------------
$ bin/prettier.js --plugin test.js --parser test test.js --print-width 120
aligned----------------------------------------------------------------------------------------------------
aligned----------------------------------------------------------------------------------------------------
aligned----------------------------------------------------------------------------------------------------
aligned----------------------------------------------------------------------------------------------------
aligned----------------------------------------------------------------------------------------------------
```

Reply:
Thank you for your reply, I will modify it as soon as possible.

Reply:
I'm not sure what's happening, it should work based on the experiment in https://github.com/prettier/prettier/pull/5701#discussion_r245212447. Could you push the changes here so that I can investigate what's happening?

Reply:
look my error commit

Reply:
thank you for your help